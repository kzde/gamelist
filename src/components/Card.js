import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import {
  margins, cardSize, colorPalettes, textSizes, themeColors
} from '../utils/styleUtils';
import ClickableText from './ClickableText';

const Wrapper = styled.div`
  width: ${props => props.width || '100%'};
  height: ${cardSize.height};
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  border-radius: 8px;
  overflow: hidden;
  opacity: 0.7;
  cursor: pointer;
  transition: opacity 0.3s;
  &:hover {
    opacity: 1;
  }
`;
const Img = styled.img`
  width: 100%;
  height: 411px;
  display: block;
`;
const Content = styled.div`
  height: 81px;
  background: ${colorPalettes.white};
  padding: ${margins.medium};
`;
const Title = styled.div`
  font-size: ${textSizes.normalText};
`;
const SubTitle = styled.div`
  margin-top: ${margins.small};
  font-size: ${textSizes.smallText};
  font-weight: 300;
`;

const Card = (props) => {
  const navTo = () => {
    props.history.push(`/${props.title}`);
  };

  return (
    <Wrapper width={props.width} onClick={navTo}>
      <Img src={props.cover} />
      <Content>
        <Title>
          <ClickableText text={props.title} handleClick={navTo} textColor={themeColors.link} />
        </Title>
        <SubTitle>{props.subTitle}</SubTitle>
      </Content>
    </Wrapper>
  );
};

Card.propTypes = {
  cover: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
  width: PropTypes.string,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired
};

export const card = Card;
export default withRouter(Card);
