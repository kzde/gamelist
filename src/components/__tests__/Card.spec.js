import React from 'react';
import renderer from 'react-test-renderer';
import { card as Card } from '../Card';

describe('Card', () => {
  it('should renders correctly', () => {
    const props = {
      cover: 'cover',
      title: 'card title',
      subTitle: 'sub title'
    };
    const tree = renderer.create(<Card {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
