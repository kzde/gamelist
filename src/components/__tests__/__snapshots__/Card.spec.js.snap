// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should renders correctly 1`] = `
<div
  className="Card__Wrapper-sc-1gjobka-0 cyrRzz"
  onClick={[Function]}
>
  <img
    className="Card__Img-sc-1gjobka-1 fekier"
    src="cover"
  />
  <div
    className="Card__Content-sc-1gjobka-2 bTjVxu"
  >
    <div
      className="Card__Title-sc-1gjobka-3 hkmeWs"
    >
      <div
        className="Link__Wrapper-sc-1kz7l5q-0 jgxdgC"
        onClick={[Function]}
      >
        <span
          className="Link__Content-sc-1kz7l5q-1 gboxtL"
          data-hover="card title"
        >
          card title
        </span>
      </div>
    </div>
    <div
      className="Card__SubTitle-sc-1gjobka-4 dnoUZn"
    >
      sub title
    </div>
  </div>
</div>
`;
